File: hw/pastor/u3/hw35CrudStrategy/CrudStrategyPY/utils/id_generator.py
from model.strategy.json_strategy import JsonStrategy
from model.strategy.csv_strategy import CsvStrategy
from model.strategy.mongo_strategy import MongoStrategy

class IdGenerator:
    @staticmethod
    def generate_unique_id() -> int:
        json_strategy = JsonStrategy()
        csv_strategy = CsvStrategy()
        mongo_strategy = MongoStrategy()
        
        json_customers = json_strategy.read_all()
        csv_customers = csv_strategy.read_all()
        mongo_customers = mongo_strategy.read_all()
        
        max_id = 0
        
        for customer in json_customers:
            if customer.id > max_id:
                max_id = customer.id
        
        for customer in csv_customers:
            if customer.id > max_id:
                max_id = customer.id
        
        for customer in mongo_customers:
            if customer.id > max_id:
                max_id = customer.id
        
        # Return next available ID
        return max_id + 1

File: hw/aymacana/u3/hw35CrudStrategy/CrudStrategyJS/utils/IdGenerator.js
import { JsonStrategy } from '../models/strategies/JsonStrategy.js';
import { CsvStrategy } from '../models/strategies/CsvStrategy.js';
import { MongoStrategy } from '../models/strategies/MongoStrategy.js';

export class IdGenerator {
    static async generateUniqueId() {
        const jsonStrategy = new JsonStrategy();
        const csvStrategy = new CsvStrategy();
        const mongoStrategy = new MongoStrategy();
        
        const jsonCustomers = await jsonStrategy.readAll();
        const csvCustomers = await csvStrategy.readAll();
        const mongoCustomers = await mongoStrategy.readAll();
        
        let maxId = 0;
        
        for (const customer of [...jsonCustomers, ...csvCustomers, ...mongoCustomers]) {
            if (customer.id > maxId) {
                maxId = customer.id;
            }
        }
    
        return maxId + 1;
    }
}